### PR TITLE
add ARCHI NEWSAN board

### PR DESCRIPTION
**Pull request by EricAlmendras**

### CODE REVIEW

File: ports/raspberrypi/boards/archi/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#define DEFAULT_UART_BUS_RX (&pin_GPIO9)
+
+#define DEFAULT_SPI_BUS_SCK (&pin_GPIO6)
+#define DEFAULT_SPI_BUS_MOSI (&pin_GPIO7)
+#define DEFAULT_SPI_BUS_MISO (&pin_GPIO4)
+
+#define MICROPY_HW_NEOPIXEL         (&pin_GPIO24)
+#define MICROPY_HW_NEOPIXEL_COUNT   (64)
+
+#define BOARD_HAS_CRYSTAL 1
```
Comment (dhalbert) [inline ports/raspberrypi/boards/archi/mpconfigboard.h:24]: This is not used for RP2040 boards.
```suggestion
```

File: ports/raspberrypi/boards/archi/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+CHIP_FAMILY = rp2
+
+EXTERNAL_FLASH_DEVICES = "W25Q32JVxQ"
+
+CIRCUITPY__EVE = 1
+CIRCUITPY_PICODVI = 1
+
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_MPU6050
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_Pixel_Framebuf
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_Led_Animation
```
Comment (dhalbert) [inline ports/raspberrypi/boards/archi/mpconfigboard.mk:16]: You'll need to add these as frozen libraries to .gitmodules. Do a `git submodule add`.
Note that for `Adafruit_CircuitPython_LED_Animation`, the LED is all capital letters.
You had `NeoPixel` here before. Did you want to add it back? It's already in `frozen/`.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Hi, one minor thing, and one comment about the frozen modules.

File: ports/raspberrypi/boards/archi/mpconfigboard.mk (hunk 2 above)
Comment (EricAlmendras) [reply ports/raspberrypi/boards/archi/mpconfigboard.mk]: I have already corrected it

Comment (EricAlmendras) [reply ports/raspberrypi/boards/archi/mpconfigboard.mk]: we take out the Adafruit_CircuitPython_Led_Animation

Comment (dhalbert) [reply ports/raspberrypi/boards/archi/mpconfigboard.mk]: Sure. It is still in `.gitmodules` in the latest push, though. Also did you want NeoPixel in or out?

**Review by dhalbert — CHANGES_REQUESTED** [review]
You changed copyrights in the files in `ports/raspberrypi/boards/odt_rpga_feather`. But that is not your board. Did you mean to do that?

**Comment by EricAlmendras** [comment]
sorry, I didn't want to change the file copyright in ports/raspberrypi/boards/odt_rpga_feather, I wanted to change it in my files, corrected.

File: .gitmodules [hunk 3]
```diff
@@ -386,3 +386,12 @@
...
 	url = https://github.com/adafruit/Adafruit_CircuitPython_Bitmap_Font
+[submodule "frozen/Adafruit_CircuitPython_MPU6050"]
+	path = frozen/Adafruit_CircuitPython_MPU6050
+	url = https://github.com/adafruit/Adafruit_CircuitPython_MPU6050
+[submodule "frozen/Adafruit_CircuitPython_Pixel_Framebuf"]
+	path = frozen/Adafruit_CircuitPython_Pixel_Framebuf
+	url = https://github.com/adafruit/Adafruit_CircuitPython_Pixel_Framebuf
+[submodule "frozen/Adafruit_CircuitPython_Led_Animation"]
+	path = frozen/Adafruit_CircuitPython_Led_Animation
+	url = https://github.com/adafruit/Adafruit_CircuitPython_Led_Animation
```
[the remark this answers is not in the document]
Comment (dhalbert) [reply .gitmodules:397]: reminder to make this change too

[the remark this answers is not in the document]
Comment (dhalbert) [reply .gitmodules:397]: Also, it doesn't look like the LED_Animation library was actually added to `frozen/`

File: ports/raspberrypi/boards/archi/mpconfigboard.mk [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+CHIP_VARIANT = RP2040
+CHIP_FAMILY = rp2
+
+EXTERNAL_FLASH_DEVICES = "W25Q32JVxQ"
+
+CIRCUITPY__EVE = 1
+CIRCUITPY_PICODVI = 1
+
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_MPU6050
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_Pixel_Framebuf
```
Comment (dhalbert) [inline ports/raspberrypi/boards/archi/mpconfigboard.mk:15]: Pixel_Framebuf depends on the LED_Animation and NeoPixel libraries, so you'll want to freeze those too, if you don't want to have the user have to add those libraries to lib/.

Alternatively, don't bother to freeze any of these, and have the user load the libraries.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks for fixing the copyrights. I still have some remaining remarks about the frozen libraries.

File: ports/raspberrypi/boards/archi/mpconfigboard.mk (hunk 4 above)
Comment (EricAlmendras) [reply ports/raspberrypi/boards/archi/mpconfigboard.mk:15]: Done. Add the LED_Animation module correcting Led by LED

File: ports/raspberrypi/boards/archi/mpconfigboard.mk [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_LED_Animation
```
Comment (dhalbert) [inline ports/raspberrypi/boards/archi/mpconfigboard.mk:16]: You didn't add NeoPixel:
```suggestion
FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_NeoPixel
```

File: .gitmodules [hunk 6]
```diff
@@ -386,3 +386,15 @@
...
```
Comment (dhalbert) [inline .gitmodules:397]: Now there are two listings for LED animation. You want to remove this one.
```suggestion
```

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/raspberrypi/boards/archi/mpconfigboard.mk (hunk 5 above)
Comment (EricAlmendras) [reply ports/raspberrypi/boards/archi/mpconfigboard.mk]: done.

File: .gitmodules (hunk 6 above)
Comment (EricAlmendras) [reply .gitmodules]: done.

**Comment by EricAlmendras** [comment]
Sorry, I added LED_Animation again, I had removed it.

**Review by dhalbert — APPROVED** [review]
OK, I think we're all set here. Thanks for your perseverance!

**Comment by dhalbert** [comment]
Could you also submit a board description to https://github.com/adafruit/circuitpython-org ? Guide to doing that is here: https://learn.adafruit.com/how-to-add-a-new-board-to-the-circuitpython-org-website

**Comment by EricAlmendras** [comment]
I will, thank you very much.

**Pull request merged**